core/unit_test/CMakeLists.txt

```diff
@@ -609,6 +609,12 @@ if(Kokkos_ENABLE_HIP)
       UnitTestMain.cpp
       hip/TestHIP_InterOp_Streams.cpp
   )
+  KOKKOS_ADD_EXECUTABLE_AND_TEST(
+    UnitTest_HIPMemoryRequirements
+    SOURCES
+      UnitTestMain.cpp
+      hip/TestHIP_Memory_Requirements.cpp
+  )
```

Comment:
Why are we adding a new executable for that test rather than adding it to an existing one?

Reply:
The existing one already has a high compile time. We should think about splitting it into several as done for other backends ... but I think this is something for another pr. Also this is kind of a different test, as it does mainly test if the allocations (+MemAdvise) we get from hip are as we expect them to be...it can be viewed as testing the requirements we build kokkos on

Reply:
For SYCL, linking is the most expensive part. Parallelizing it by splitting into multiple executables also helps with memory constraints. We also had issues on Windows with CUDA support.
In my experience, splitting eases debugging, too, (and allows running tests in parallel). Just my 2¢.

Reply:
Please change and add to the main exe for consistency.  This can be revisited later as necessary.

Reply:
moved into main tests in [ab4d0c](https://github.com/kokkos/kokkos/pull/5112/commits/ab4d0cb12d033b60531c68a878e7343f28b8c7fc)